### PR TITLE
use c++11 shared_ptr which is used in newer urdf versions

### LINKED ISSUE #181: cob_omni_drive_controller fails to compile with newer urdf versions
Compilation fails because newer urdf uses c++11 shared_ptr for urdf::JointConstSharedPtr

```
/var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp: In function ‘bool parseWheelGeom(WheelGeom&, XmlRpc::XmlRpcValue&, MergedXmlRpcStruct&, urdf::Model*)’:
/var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp:113:54: error: no match for ‘operator=’ (operand types are ‘boost::shared_ptr<const urdf::Joint>’ and ‘urdf::JointConstSharedPtr {aka std::shared_ptr<const urdf::Joint>}’)
         steer_joint = model->getJoint(geom.steer_name);
                                                      ^
In file included from /usr/include/boost/shared_ptr.hpp:17:0,
                 from /opt/ros/kinetic/include/ros/forwards.h:37,
                 from /opt/ros/kinetic/include/ros/common.h:37,
                 from /opt/ros/kinetic/include/ros/ros.h:43,
                 from /var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/include/cob_omni_drive_controller/UndercarriageCtrlGeomROS.h:21,
                 from /var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp:18:
```

@fmessmer FYI

### PR DESCRIPTION
**Pull request by pholthaus**
- fixes #181
 - no new files added

### CODE REVIEW

**Comment by fmessmer** [comment]
do you need to set `set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")`

**Comment by pholthaus** [comment]
Sorry, my build system forces c++11, so I have missed that.

File: cob_omni_drive_controller/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(cob_omni_drive_controller)
 
+set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
+
```
Comment (mgruhler) [inline cob_omni_drive_controller/CMakeLists.txt]: @ipa-fxm `catkin_lint` will complain about this. Manually changing `CMAKE_CXX_FLAGS` should be avoided.

Use `add_compile_options(-std=c++11)` instead.

Comment (fmessmer) [reply cob_omni_drive_controller/CMakeLists.txt]: thanks, @mgruhler for still caring about this repo
I knew it was something shorter ;-)

@pholthau could you make the according changes to the two affected pkgs?
I will then squash-merge...

Comment (mgruhler) [reply cob_omni_drive_controller/CMakeLists.txt]: For sure! I still geht the notifications :-)

**Comment by fmessmer** [comment]
I took the liberty to set the `compile_option` as suggested...still Travis fails for `std::shared_ptr` vs. `boost::shared_ptr`...but `kinetic_devel` passes **as is**...

what version of urdf are you using? is there a difference between ubuntu and gentoo? what's your ROS_DISTRO?

**Comment by pholthaus** [comment]
Sorry, I am away from the computer most of the time until Wednesday. Thank you for adding the options.
Gentoo's urdf version is ```ros-kinetic/urdf-1.12.12```
```ROS_DISTRO``` is ```kinetic```

**Comment by pholthaus** [comment]
Seems to be the same in ubuntu (cf. [jenkins](http://build.ros.org/view/Kbin_uX64/job/Kbin_uX64__urdf__ubuntu_xenial_amd64__binary/44/)), so maybe something goes wrong due to the fact that packages are compiled during installation (and ```-std=c++11``` is **default**). Can anybody from the overlay comment on this maybe @allenh1?

**Comment by allenh1** [comment]
@pholthau I'm very puzzled by this one -- the UDRF versions are the same, and there are no patches in the overlay at the moment for that package (so the same code is made for both).

I'd suggest that the `auto` keyword be used here, so that it could work with either. To my knowledge, they have essentially the same interface.

**Comment by allenh1** [comment]
Please disregard that statement -- that makes no sense in the context of this patch (I had not looked at the delta yet, just the issue).

I'll have to look at this on my machine later, as I'm not around it at the moment. I'll write back when I figure something out.

**Comment by fmessmer** [comment]
lets have a look how ros_control recently fixed these issues...and follow their lead...:wink:

**Comment by fmessmer** [comment]
superseded by https://github.com/ipa320/cob_control/pull/208

**Pull request closed without merging**